packages/open-api-gateway/src/project/open-api-gateway-python-project.ts

```diff
@@ -112,12 +118,9 @@ export class OpenApiGatewayPythonProject extends PythonProject {
     this.apiSrcDir = options.apiSrcDir ?? "api";
 
     // Generated project should have a dependency on this project, in order to run the generation scripts
-    [
-      OPENAPI_GATEWAY_PDK_PACKAGE_NAME,
-      "constructs",
-      "aws-cdk-lib",
-      "cdk-nag",
-    ].forEach((dep) => this.addDependency(dep));
+    [OPENAPI_GATEWAY_PDK_PACKAGE_NAME, "constructs", "aws-cdk-lib", "cdk-nag"]
+      .filter((dep) => !this.deps.tryGetDependency(dep, DependencyType.RUNTIME))
```

Comment:
What about pinned versions? Do we need to split the dep on @ here as well?

Reply:
The list of versions we're filtering doesn't have any `@`s in it so I omitted the splitting :)